core/src/worker/workflow/machines/workflow_machines.rs

```diff
@@ -420,9 +418,15 @@ impl WorkflowMachines {
                 Some(workflow_activation_job::Variant::QueryWorkflow(_))
             )
         });
+        let is_replaying = self.replaying || all_query;
```

Comment:
Aren't queries considered replaying anyway?

Reply:
Not until this point, that's where it happens